cartography/intel/gcp/apigateway.py

```diff
@@ -164,20 +190,33 @@ def get_gateways(apigateway: Resource, project_id: str) -> List[Dict]:
     """
     gateways = []
     try:
-        req = apigateway.projects().locations().gateways().list(parent=f'projects/{project_id}/locations/*')
-        while req is not None:
-            res = req.execute()
-            if res.get('gateways', []):
-                for gateway in res['gateways']:
-                    gateway['id'] = gateway['name']
-                    gateway['project_id'] = project_id
-                    x = gateway.get('name').split('/')
-                    x = x[x.index('locations') + 1].split("-")
-                    gateway['region'] = x[0]
-                    if len(x) > 1:
-                        gateway['region'] = f"{x[0]}-{x[1]}"
-                    gateways.append(gateway)
-            req = apigateway.projects().locations().gateways().list_next(previous_request=req, previous_response=res)
+        for location in locations:
+            req = apigateway.projects().locations().gateways().list(parent=f'projects/{project_id}/locations/{location["id"]}')
+            while req is not None:
+                res = req.execute()
+                if res.get('gateways', []):
+                    for gateway in res['gateways']:
+                        gateway['id'] = gateway['name']
+                        gateway['project_id'] = project_id
+                        gateway['iam_policy'] = apigateway.projects().locations().gateways().getIamPolicy(\
+                            resource = f"projects/{project_id}/locations/{location['id']}/gateways/{gateway['name']}")
+                        bindings = gateway.get['iam_policy',{}].get['bindings',[]]
+                        members = []
+                        for binding in bindings:
+                            members.append(binding.get('members',[]))
+                        gateway['members'] = members
+                        for member in gateway['members']:
```

Comment:
here also. similar to compute and kms, we should build relationship between iam resources and apigateway.